src/MoveUntilTouchTopicController.cpp

```diff
@@ -0,0 +1,218 @@
+#include <rewd_controllers/MoveUntilTouchTopicController.hpp>
+
+#include <functional>
+#include <pluginlib/class_list_macros.h>
+
+namespace rewd_controllers
+{
+//=============================================================================
+MoveUntilTouchTopicController::MoveUntilTouchTopicController()
+    : MultiInterfaceController{true}  // allow_optional_interfaces
+    , JointTrajectoryControllerBase{}
+    , mTaringCompleted{false}
+    , mForceThreshold{0.0}
+    , mTorqueThreshold{0.0}
+{
+}
+
+//=============================================================================
+MoveUntilTouchTopicController::~MoveUntilTouchTopicController() {}
+
+//=============================================================================
+bool MoveUntilTouchTopicController::init(hardware_interface::RobotHW* robot,
+                                    ros::NodeHandle& nh)
+{
+  // check that doubles are lock-free atomics
+  if (!mForceThreshold.is_lock_free()) {
+    ROS_ERROR(
+        "double atomics not lock-free on this system. Cannot guarantee "
```

Comment:
Nit: "double ... "-> "Double ... "

Shouldn't we check for both `mForceThreshold` and `mTorqueThreshold`?

Reply:
My understanding is that on one system either all atomics are lock free or none of them are, so checking one should suffice.